src/vmq_reg.erl

```diff
@@ -634,18 +628,18 @@ del_subscriptions(Topics, SubscriberId) ->
 maybe_remap_subscriber(SubscriberId, #{clean_session := true}) ->
     %% no need to remap, we can delete this subscriber
     %% we overwrite any other value
-    Subs = vmq_subscriber:new(),
+    Subs = vmq_subscriber:new(true),
     vmq_subscriber_db:store(SubscriberId, Subs),
     {false, Subs, []};
 maybe_remap_subscriber(SubscriberId, _) ->
     case vmq_subscriber_db:read(SubscriberId) of
         undefined ->
             %% Store empty Subscriber Data
-            Subs = vmq_subscriber:new(),
+            Subs = vmq_subscriber:new(false),
```

Comment:
Maybe it would be nicer to match the clean session flag from the map and pass it on to the `vmq_subscriber:new/1` here and above as well? I was looking at the code and had to think twice why it was true/false in the two cases.